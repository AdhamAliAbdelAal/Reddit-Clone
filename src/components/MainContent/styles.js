import { styled } from '@mui/system';

export const MainContentContainer = styled('div')(({ theme, width }) => ({
  [theme.breakpoints.up('md')]: {
    width,
  },
  [theme.breakpoints.down('md')]: {
    width: '100%',
  },
}));
