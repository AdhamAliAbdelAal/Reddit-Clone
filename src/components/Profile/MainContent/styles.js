import { Box, styled } from '@mui/material';

export const ProfilePage = styled(Box)(({ theme }) => ({
  minHeight: '100vh',
  maxWidth: '100%',
  backgroundColor: '#dae0e6',
  display: 'flex',
  justifyContent: 'center',
  padding: '20px 24px',

  [theme.breakpoints.between('0', '650')]: {
    padding: 0,
  },
}));

export const NEW = styled(Box)(() => ({
  minHeight: 76,
  backgroundColor: 'white',
  border: '1px solid #edeff1',
  borderRadius: 4,
  marginBottom: 10,
  padding: 16,
  display: 'flex',
  alignItems: 'center',

}));
export const NewBox = styled(Box)(() => ({

  boxShadow: '0 0 8px #6a5cff66',
  width: '100%',
  display: 'flex',
  justifyContent: 'space-between',
  color: '#1c1c1c',
  fontWeight: 500,
  fontSize: 16,
  padding: 12,
  borderRadius: 8,

}));

export const WideBox = styled(Box)(() => ({
  width: '100%',
}));
