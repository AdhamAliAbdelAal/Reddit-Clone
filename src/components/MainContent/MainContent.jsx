import { MainContentContainer } from './styles';
/**
 * This component works as a container for the left half components
 * like timeline
 *
 * @component MainContent
 * @property {React.Component} children -Components which will be wrapped in main container
 * @property {number} width -The maximum width of the main content
 * @returns {React.Component} Container represents its children
 */

function MainContent({ children, width }) {
  return (
    <MainContentContainer width={width}>
      {children}
    </MainContentContainer>
  );
}

export default MainContent;
