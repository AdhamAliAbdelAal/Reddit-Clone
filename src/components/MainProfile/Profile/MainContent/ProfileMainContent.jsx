import { Box } from '@mui/material';
import { useParams } from 'react-router-dom';
import CommunitiesSubscriberProvider from '../../../../contexts/CommunitiesSubscriberContext';
import CommunitiesProvider from '../../../../contexts/CommunitiesModeratorContext';
import Content from './Content/Content';
import FollowersList from './FollowersList/FollowersList';
import HistoryTap from './HistoryTap/HistoryTap';
import PostsFilteredTap from './PostsFilteredTap/PostsFilteredTap';
import PostsTap from './PostsTap/PostsTap';
import SavedTap from './SavedTap/SavedTap';
import Sidebar from './Sidebar/Sidebar';
import { ProfilePage } from './styles';
import CommentsTap from '../../CommentsTap/CommentsTap';
import UserProvider from '../../../../contexts/UserProvider';

const renderSwitch = (param) => {
  if (param === undefined || param === '?sort=new' || param === '?sort=hot' || param === '?sort=top' || param === '?sort=top&t=day') {
    return (
      <Content />
    );
  }

  if (param === 'submitted' || param === 'submitted/?sort=new') {
    return (
      <PostsTap />
    );
  }
  if (param === 'upvoted' || param === 'downvoted' || param === 'hidden') {
    return (
      <PostsFilteredTap />
    );
  }
  if (param === 'saved') {
    return (
      <SavedTap />
    );
  }
  if (param === 'history') {
    return (
      <HistoryTap />
    );
  }
  if (param === 'comments') {
    return (
      <CommentsTap profile />
    );
  }
  if (param === 'followers') {
    return (
      <FollowersList />
    );
  }
  return (
    <div />
  );
};

/**
 * MainContent if the profile page
 *
 * @component MainContentProfile
 * @property  {function} handleClick to navigate to the subpage
 * @return {React.Component} - MainContent
 */

function ProfileMainContent() {
  const { subTitle, username } = useParams();
  return (
    <Box>
      <UserProvider name={username}>
        <CommunitiesProvider>
          <CommunitiesSubscriberProvider>
            <ProfilePage>
              {renderSwitch(subTitle)}
              <Sidebar />
            </ProfilePage>
          </CommunitiesSubscriberProvider>
        </CommunitiesProvider>
      </UserProvider>

    </Box>
  );
}

export default ProfileMainContent;
