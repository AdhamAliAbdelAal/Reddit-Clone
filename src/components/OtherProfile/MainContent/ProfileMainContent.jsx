import { Box } from '@mui/material';
import { useParams } from 'react-router-dom';
import ContentProvider from '../../../context/ContentProvider';
import PostsProvider from '../../../context/PostsProvider';
import UserProvider from '../../../context/UserProvider';
import UserInfoProvider from '../../../context/UserInfoProvider';
import Content from './Content/Content';
import PostsTap from './PostsTap/PostsTap';
import Sidebar from './Sidebar/Sidebar';
import { ProfilePage } from './styles';

const renderSwitch = (param, username) => {
  if (param === undefined || param === 'sort=new' || param === 'sort=hot' || param === 'sort=top' || param === 'sort=top&t=day') {
    return (
      <ContentProvider name={username}>
        <Content />
      </ContentProvider>
    );
  }

  if (param === 'submitted') {
    return (
      <PostsProvider name={username}>
        <PostsTap />
      </PostsProvider>
    );
  }
  return (
    <div />
  );
};

/**
 * MainContent if the profile page
 *
 * @component MainContentProfile
 * @property  {function} handleClick to navigate to the subpage
 * @return {React.Component} - MainContent
 */

function ProfileMainContent() {
  const { subTitle, username } = useParams();
  return (
    <Box>
      <UserProvider name={username}>
        <ProfilePage>
          {renderSwitch(subTitle, username)}
          <UserInfoProvider name={username}>
            <Sidebar />
          </UserInfoProvider>
        </ProfilePage>
      </UserProvider>
    </Box>
  );
}

export default ProfileMainContent;
