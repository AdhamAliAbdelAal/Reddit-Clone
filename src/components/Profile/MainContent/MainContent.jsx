import { Box } from '@mui/material';
import { useParams } from 'react-router-dom';
import FilteredPostsProvider from '../../../context/FilteredPostsProvider';
import ContentProvider from '../../../context/ContentProvider';
import PostsProvider from '../../../context/PostsProvider';
import UserProvider from '../../../context/UserProvider';
import Content from './Content/Content';
import PostsFilteredTap from './PostsFilteredTap/PostsFilteredTap';
import PostsTap from './PostsTap/PostsTap';
import Sidebar from './Sidebar/Sidebar';
import { ProfilePage } from './styles';

const renderSwitch = (param) => {
  if (param === undefined || param === 'sort=new' || param === 'sort=hot' || param === 'sort=top' || param === 'sort=top&t=day') {
    return (
      <ContentProvider>
        <Content />
      </ContentProvider>
    );
  }

  if (param === 'submitted' || param === 'submitted/sort=new') {
    return (
      <PostsProvider>
        <PostsTap />
      </PostsProvider>
    );
  }
  if (param === 'upvoted' || param === 'downvoted' || param === 'saved' || param === 'hidden') {
    return (
      <FilteredPostsProvider param={param}>
        <PostsFilteredTap type={param} />
      </FilteredPostsProvider>
    );
  }
  return (
    <>no</>
  );
};

/**
 * Main content of the profile pagee containing comments and posts
 * @return {React.Component} - MainContent
 */
function MainContent() {
  const { subTitle } = useParams();

  return (
    <Box>
      <UserProvider>
        <ProfilePage>
          {renderSwitch(subTitle)}
          <Sidebar />
        </ProfilePage>
      </UserProvider>
    </Box>
  );
}

export default MainContent;
