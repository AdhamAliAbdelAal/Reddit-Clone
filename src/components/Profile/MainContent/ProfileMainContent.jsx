import { Box } from '@mui/material';
import { useParams } from 'react-router-dom';
import FilteredPostsProvider from '../../../contexts/FilteredPostsProvider';
import ContentProvider from '../../../contexts/ContentProvider';
import PostsProvider from '../../../contexts/PostsProvider';
import UserProvider from '../../../contexts/UserProvider';
import UserInfoProvider from '../../../contexts/UserInfoProvider';
import Content from './Content/Content';
import PostsFilteredTap from './PostsFilteredTap/PostsFilteredTap';
import PostsTap from './PostsTap/PostsTap';
import Sidebar from './Sidebar/Sidebar';
import { ProfilePage } from './styles';

const renderSwitch = (param, username) => {
  if (param === undefined || param === 'sort=new' || param === 'sort=hot' || param === 'sort=top' || param === 'sort=top&t=day') {
    return (
      <ContentProvider name={username}>
        <Content />
      </ContentProvider>
    );
  }

  if (param === 'submitted' || param === 'submitted/sort=new') {
    return (
      <PostsProvider name={username}>
        <PostsTap />
      </PostsProvider>
    );
  }
  if (param === 'upvoted' || param === 'downvoted' || param === 'saved' || param === 'hidden') {
    return (
      <FilteredPostsProvider param={param} name={username}>
        <PostsFilteredTap type={param} />
      </FilteredPostsProvider>
    );
  }
  return (
    <div />
  );
};

/**
 * MainContent if the profile page
 *
 * @component MainContentProfile
 * @property  {function} handleClick to navigate to the subpage
 * @return {React.Component} - MainContent
 */

function ProfileMainContent() {
  const { subTitle, username } = useParams();
  return (
    <Box>
      <UserProvider name={username}>
        <ProfilePage>
          {renderSwitch(subTitle, username)}
          <UserInfoProvider name={username}>
            <Sidebar />
          </UserInfoProvider>
        </ProfilePage>
      </UserProvider>
    </Box>
  );
}

export default ProfileMainContent;
