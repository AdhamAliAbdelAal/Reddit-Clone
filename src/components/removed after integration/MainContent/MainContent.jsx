import { MainContentContainer } from './styles';

function MainContent({ children }) {
  return (
    <MainContentContainer>
      {children}
    </MainContentContainer>
  );
}

export default MainContent;
