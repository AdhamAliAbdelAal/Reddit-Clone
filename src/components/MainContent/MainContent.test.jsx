import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import MainContent from './MainContent';

describe(MainContent, () => {
  it('MainContent renders correctly', () => {
    const tree = renderer
      .create(
        <MainContent width={200}>
          <div>dummy</div>
        </MainContent>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
