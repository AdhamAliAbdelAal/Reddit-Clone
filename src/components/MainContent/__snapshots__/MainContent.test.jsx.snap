// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContent MainContent renders correctly 1`] = `
<div
  className="css-xd4h8n"
  width={200}
>
  <div>
    dummy
  </div>
</div>
`;
